src/IECoreImage/DisplayDriverServer.cpp

```diff
@@ -99,16 +122,51 @@ class DisplayDriverServer::PrivateData : public RefCounted
 
 		PrivateData( int portNumber ) :
 			m_success(false),
-			m_endpoint(tcp::v4(), portNumber),
 			m_service(),
 			m_acceptor( m_service ),
 			m_thread()
 		{
-			m_acceptor.open(  m_endpoint.protocol() );
-			m_acceptor.set_option( boost::asio::ip::tcp::acceptor::reuse_address(true));
-			m_acceptor.bind( m_endpoint );
-			m_acceptor.listen();
-			m_success = true;
+			if( portNumber == 0 && g_portRange[0] != std::numeric_limits<int>::min() )
+			{
+				for( int p = g_portRange[0]; p < g_portRange[1] + 1; ++p )
+				{
+					try
+					{
+						m_endpoint = boost::asio::ip::tcp::endpoint( tcp::v4(), p );
+						m_acceptor.open(  m_endpoint.protocol() );
+						m_acceptor.set_option( boost::asio::ip::tcp::acceptor::reuse_address(true));
+						m_acceptor.bind( m_endpoint );
+						m_acceptor.listen();
+						m_success = true;
+						break;
+					}
+					catch( ... )
+					{
+						m_acceptor.close();
+					}
+				}
+
+				if( !m_success )
+				{
+					throw IECore::IOException( "DisplayDriverServer : Unable to find a free port in the range [ " + std::to_string( g_portRange[0] ) + ", " + std::to_string( g_portRange[1] ) + " ]." );
+				}
+			}
+			else
+			{
+				try
+				{
+					m_endpoint = boost::asio::ip::tcp::endpoint( tcp::v4(), portNumber );
+					m_acceptor.open( m_endpoint.protocol() );
+					m_acceptor.set_option( boost::asio::ip::tcp::acceptor::reuse_address( true ) );
+					m_acceptor.bind( m_endpoint );
+					m_acceptor.listen();
```

Comment:
There are enough funky little details here that I think it might be good to avoid duplicating the code. Maybe a little private `openPort()` method to encapsulate it all?

Reply:
Addressed in a5958ff (note that I squashed changes into the original commit)